gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+ *
+ * Note: While the JAWS solution entry is fairly stable, if there are changes to the settings
+ * configuration names, this will need to be updated.
+ *
+ * @param {Object} handlerSpec - The solutions registry entry for JAWS
+ * @param {Function} localResolver - The localResolver, usually from the `userSession` on the
+ * `lifecycleManager` to resolve the INI configuration paths where JAWS stores it's settings.
+ * @return {Object} Returns a new handlerSpec with the configurations adjusted depending on the
+ * current default voice profile on the system. This is a deep copy of `handlerSpec` the original
+ * is not modified.
+ */
+gpii.flowManager.jawsVoiceProfileAdjustment = function (handlerSpec, localResolver) {
+    var togo = fluid.copy(handlerSpec);
+
+    // Configuration1 contains the path to the default settings stored in INI format
+    var defaultJCFpath = localResolver(handlerSpec.settingsHandlers.configuration1.options.filename);
+    var defaultJAWsSettings = gpii.iniFile.readFile(defaultJCFpath, {});
+
+    // It's possible to not have a current voice profile set in DEFAULT.JCF
+    // In the DEFAULT.JCF file it is stored here:
+    // ```ini
+    // [Voice Profiles]
+    // ActiveVoiceProfileName=GPII
+    // ```
+    var currentVoiceProfile = false;
+    if (defaultJAWsSettings["Voice Profiles"] && defaultJAWsSettings["Voice Profiles"].ActiveVoiceProfileName) {
+        currentVoiceProfile = defaultJAWsSettings["Voice Profiles"].ActiveVoiceProfileName;
+    }
+
+    // Again this will need to be updated in sync with JAWS support. As of now, we don't store this path by itself in the
+    // solution entry.
+    var voiceProfilesDir = localResolver("${{environment}.APPDATA}\\Freedom Scientific\\JAWS\\2019\\Settings\\VoiceProfiles\\");
+    var voiceProfilesPath = voiceProfilesDir + currentVoiceProfile + ".VPF";
+    var voiceProfileExists = fs.existsSync(voiceProfilesPath);
+
+    if (!voiceProfileExists) {
+        // If the voice profile does not exist delete the second configuration from the
+        // handlerSpec. The INI settings handler will fail for both configurations if the
+        // file is missing.
+        delete togo.settingsHandlers.configuration2;
+    }
+    else {
+        // Otherwise update the .VPC ini configuration with the voice profile path
+        togo.settingsHandlers.configuration2.options.filename = voiceProfilesPath;
+    }
+
+    return togo;
+};
+
+/**
+ * This makes adjustments (hacks really) for the return payload of certain settings handlers,
+ * which when invoked return the value of the setting in a separate sub-structure, rather than
+ * as the direct proper of the settings block.  This should be temporary, and we should resolve
+ * why the SPI and Native Settings Handler snapshots come back as (technically) generic capability
+ * versions, rather than the transformed app specific settings like the rest of the settings
+ * handlers.
+ *
+ *     "com.microsoft.windows.magnifier": [
```

Comment:
Please describe what these examples are?

Reply:
this has been removed

Reply:
This section has been removed